### ISSUE TITLE
List plugins inside plugin packs

### ISSUE THREAD

**Issue by ben-eb** [himynameisdave/postcss-plugins#21]
It would be good to list plugins as parts of plugin packs. So for example cssnext uses postcss-selector-matches, and at least a handful of plugins use postcss-reporter. Would be great if those connections were realised in this list. :smile:

Would be nice also to list some plugins that are not included, once this is done. We decided to remove most of cssnano's plugins (there are a lot now!) from the PostCSS readme on account of the optimisations section being too big.

**Comment by himynameisdave** [comment]
Yeah this is actually super cool! Looking to add this stuff as well!

Here's how I'm thinking I'm going to implement this:

1. Have a `docs/packs.md` file that displays each pack's dependant plugins.
2. Go research the plugins used by packs like `cssnano` and make sure they are added to the main list

@ben-eb Make sense?

**Comment by ben-eb** [comment]
You could also have a `packs` property for the plugin (array type) that would list the packs that the plugin belongs to. :smile:

**Comment by himynameisdave** [comment]
That is so much smarter/better, and then if I really wanted to generate that `packs.md` it would be so much easier because that data would be available.

**Comment by ben-eb** [comment]
Alright, I'll send a patch for cssnano's plugins. :smile:

**Comment by jonathantneal** [comment]
I would rather the packs identify which plugins they utilize. This might be inferable from package.json

**Comment by himynameisdave** [comment]
![](https://33.media.tumblr.com/0ae4a458b02b1705c96dd846493e2aad/tumblr_inline_n3ghj9J3u91qcklud.gif)

I like both of these ideas... so here's what I think is the most comprehensive:
```json
{
  "name": "cool-plugin-pack",
  "tags": [
    "pack"
  ],
 "plugins_used": [
    "a-neat-plugin",
    "other-plugin"
 ]
},{
   "name": "a-neat-plugin",
   "tags": [
     "fun"
   ],
  "packs": [
     "cool-plugin-pack"
  ]
}
```
(obviously I left out some properties but you get the idea...)

**Comment by ben-eb** [comment]
We can get the other by code, surely?

**Comment by himynameisdave** [comment]
@ben-eb how do you mean? I see it being used as such:

```javascript
require("postcss-plugins").forEach(function(plugin){
  if( plugin.tags.indexOf("pack") < 0 ){
    // this plugin is a pack
    // plugin.plugins_used exists and can be iterated through
  }else{
     if( plugin.packs ){
       // this plugin is part of a pack, and plugin.packs can be iterated through
     }else{
       // this plugin is not a pack or part of a pack
     }
  }
});
```

**Comment by ben-eb** [comment]
I mean that you can infer the relationship between the pack & each of its plugins without needing two properties, whichever you choose.

**Comment by himynameisdave** [comment]
....but what do you think is the best course of action? Tell me, I wanna know - what would you do?

![](https://magoosh.com/act/files/2015/02/giphy-1.gif)

**Comment by ben-eb** [comment]
I already started using the `packs` property in the last pull request, so I would go with that. :smile:

**Comment by jonathantneal** [comment]
Related to #13, might we use `https://registry.npmjs.org/${ name }/latest` to get the *list of dependencies* as `dependencies`? We could use this to report the number of dependencies or what those dependencies are, as well as detect when those dependencies are other PostCSS plugins.

**Comment by himynameisdave** [comment]
Well we kind of already have the `packs` property in the dataset, I'm not sure there is a ton of value in adding `dependencies` as well but maybe I'm wrong?
